### PR TITLE
Use automated integration tests

### PR DESCRIPTION
**Pull request by kbalk**
Uses tardigrade-ci's ability to run automated integration tests against a mock AWS stack.

### CODE REVIEW

File: tests/create_users_policies/main.tf [hunk 1]
```diff
@@ -168,7 +164,7 @@ locals {
     account_id    = data.aws_caller_identity.current.account_id
     partition     = data.aws_partition.current.partition
     region        = data.aws_region.current.name
-    random_string = random_string.this.result
+    random_string = "${random_string.this.result}"
```
Comment (lorengordon) [inline tests/create_users_policies/main.tf:167]: This interpolation is not necessary. Somewhat surprised `terraform fmt -recursive` is not removing it.

```suggestion
    random_string = random_string.this.result
```

File: tests/create_users_policies/main.tf [hunk 2]
```diff
@@ -189,7 +185,7 @@ locals {
 }
 
 locals {
-  test_id = data.terraform_remote_state.prereq.outputs.random_string.result
+  test_id = data.terraform_remote_state.prereq.outputs.random_string
```
Comment (lorengordon) [inline tests/create_users_policies/main.tf:188]: This is passing for me without these changes, in a local test run. I'm guessing they were from some earlier implementation and no longer relevant.

```suggestion
  test_id = data.terraform_remote_state.prereq.outputs.random_string.result
```

File: tests/create_users_policies/prereq/main.tf [hunk 3]
```diff
@@ -6,5 +6,5 @@ resource "random_string" "this" {
 }
 
 output "random_string" {
-  value = random_string.this
+  value = random_string.this.result
```
Comment (lorengordon) [inline tests/create_users_policies/prereq/main.tf:9]: Related to above comment...

```suggestion
  value = random_string.this
```

**Review by lorengordon — COMMENTED** [review]
Just a few changes to a test that don't appear necessary, otherwise ready to go

File: tests/create_users_policies/main.tf (hunk 1 above)
Comment (kbalk) [reply tests/create_users_policies/main.tf:167]: ok

File: tests/create_users_policies/main.tf (hunk 2 above)
Comment (kbalk) [reply tests/create_users_policies/main.tf:188]: ok

File: tests/create_users_policies/prereq/main.tf (hunk 3 above)
Comment (kbalk) [reply tests/create_users_policies/prereq/main.tf]: ok

**Review by lorengordon — APPROVED** [review]

**Pull request merged**